src/main/java/net/minecraftforge/event/world/BlockEvent.java

```diff
@@ -444,4 +445,55 @@ public PortalSpawnEvent(IWorld world, BlockPos pos, BlockState state, NetherPort
             return size;
         }
     }
+    
+    /**
+     * Fired when when this block is right clicked by a tool to change its state.
+     * For example: Used to determine if an axe can strip, a shovel can path, or a hoe can till.
+     * 
+     * This event is {@link Cancelable}. If canceled, this will prevent the tool
+     * from changing the block's state.
+     */
+    @Cancelable
+    public static class BlockToolInteractEvent extends BlockEvent
+    {
+
+        private final PlayerEntity player;
+        private final ItemStack stack;
+        private final ToolType toolType;
+        private BlockState state;
+
+		public BlockToolInteractEvent(IWorld world, BlockPos pos, BlockState originalState, PlayerEntity player, ItemStack stack, ToolType toolType)
+	    {
+		    super(world, pos, originalState);
+		    this.player = player;
+		    this.stack = stack;
+		    this.state = originalState;
+		    this.toolType = toolType;
+		}
+
+	    public PlayerEntity getPlayer()
+	    { 
+	        return player;
+	    }
+		
+	    public ItemStack getHeldItemStack()
+	    {
+	        return stack;
+	    }
+
+	    public ToolType getToolType()
+	    {
+	        return toolType;
+	    }
+
+	    public void setFinalState(BlockState finalState)
+	    {
+	        this.state = finalState;
+	    }
+
+	    public BlockState getFinalState()
+	    {
+	        return state;
+	    }
```

Comment:
It probably would be nice to java doc all the methods in this event, but at the very least these two methods should have an explanation of what they do.